File: CountStringFromWeb.Tests/HttpsContentTest.cs
using NUnit.Framework;

namespace CountStringFromWeb.Tests {
    [TestFixture]
    public class GetHttpsContent {
        [Test]
        public void ReceiveRightContentTest() {
            // arrange
            var tcpClient = TestHelper.GenerateSimpleTcpClientMock();

            // act
            SecureWebContentReader reader = new SecureWebContentReader(tcpClient.Object, TestHelper.Url);
            var actual = reader.GetHttpsContent();

            // assert
            StringAssert.AreEqualIgnoringCase("1234567890", actual);
        }
    }
}

File: JavaCountStringFromWeb/src/test/java/at/fhtw/bif3/swe1/countstringfromweb/HttpsContentTest.java
package at.fhtw.bif3.swe1.countstringfromweb;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class HttpsContentTest {

    @Test
    void testReceiveRightContent() {
        // arrange
        var tcpClient = TestHelper.generateSimpleTcpClientMock();

        // act
        SecureWebContentReader reader = new SecureWebContentReader(tcpClient, TestHelper.URL);
        var actual = reader.getHttpsContent();

        // assert
        assertEquals("1234567890", actual);
    }
}